### ISSUE TITLE
HTTP 103 fails in asObservableSuccess

### ISSUE THREAD

**Issue by vetleledaal** [tachiyomiorg/tachiyomi#6623]
### Describe your suggested feature

Related to #6562, but that issue is closed for comments.

asObservableSuccess() does not properly handle HTTP 103 codes. These should be ignored instead of causing the whole request to fail.

### Other details

This is difficult to test as it seems like Cloudflare is doing A/B testing on this feature, but here are some repro steps:
1. Revert #6562
2. Open GManga v1.2.9 (you _may_ need to change the UA to bypass CF (setting it to any number is sufficient))
3. Open "National School Prince is a Girl"
4. If and only if the toast "HTTP error 103" is shown on manual refresh; continue
5. Open chapter 237 OR 236 (others are not affected)
 
I used this snippet for breakpoints
```kt
override fun fetchPageList(chapter: SChapter): Observable<List<Page>> {
    return client.newCall(pageListRequest(chapter))
        .asObservable()
        .map { response ->
            pageListParse(response)
        }
}
```
![studio64_DW2aY4a4fX](https://user-images.githubusercontent.com/13540478/153489431-17bdcf9e-bb99-423c-83db-b1d0d77381c5.png)

Used a VPN to get HTTP 103 codes, Surfshark (Germany - Berlin)

Related issue tachiyomiorg/tachiyomi-extensions#10786

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by vetleledaal** [comment]
this is still happening in a131e28b6. With the official GMANGA v1.2.9 it says "HTTP error 103" when opening the mentioned chapters, switching to .asObservable() makes he toast say "expected 0 bytes but received 3463"

**Comment by vetleledaal** [comment]
The GManga extension was fixed in tachiyomiorg/tachiyomi-extensions#10934 by changing the domain to one that doesn't return 103 codes. The underlying issue is still present, but probably not as urgent as before.

**Comment by hakudansho** [comment]
Gmanga was working just fine but now there is (http error 520) now when updating, should I open new issue or it's related to this issue?

**Comment by alessandrojean** [comment]
HTTP 5xx errors are server side, nothing that can be done on Tachi side.

**Comment by arkon** [comment]
Sort of addressed by #7907, although we should clean it up if OkHttp ever handles it properly.
